### PR TITLE
nomad_1_1: 1.1.2 -> 1.1.3, nomad_0_12: drop

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
https://github.com/hashicorp/nomad/blob/v1.1.3/CHANGELOG.md

Switches to buildGoModule (for 1.1.x) as it's no longer an option not to. Also defaults to Nomad 1.1.x from now on.

See #131980 for 1.0.9 bump, opened separately to make backports easier (we do not want to drop 0.12 in 21.05).

cc @endocrimes for reviewing.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131981` at 0bab8d4f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>nomad (nomad_1_1)</li>
  </ul>
</details>

**Review by endocrimes — APPROVED** [review]

**Review by risicle — APPROVED** [review]
Builds & runs, macos 10.15 & nixos x86_64

File: pkgs/applications/networking/cluster/nomad/genericModule.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  pname = "nomad";
+  inherit version;
+  rev = "v${version}";
+
+  subPackages = [ "." ];
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = pname;
+    inherit rev sha256;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/nomad/genericModule.nix:22]:
```suggestion

  subPackages = [ "." ];

  src = fetchFromGitHub {
    owner = "hashicorp";
    repo = pname;
    rev = "v${version}";
    inherit sha256;
```

File: pkgs/applications/networking/cluster/nomad/genericModule.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  postFixup = lib.optionalString nvidiaGpuSupport ''
+    for bin in $out/bin/*; do
+      patchelf --add-needed "${nvidia_x11}/lib/libnvidia-ml.so" "$bin"
+    done
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.nomadproject.io/";
+    description = "A Distributed, Highly Available, Datacenter-Aware Scheduler";
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/nomad/genericModule.nix:56]:
```suggestion
```
default when using buildGoModule

File: pkgs/applications/networking/cluster/nomad/genericModule.nix (hunk 1 above)
Comment (maxeaubrey) [reply pkgs/applications/networking/cluster/nomad/genericModule.nix]: Done, but it is rather funny to me, as `inherit rev` came at the suggestion of another committer in the original multi-version PR - making requests for changes like this feel like a rather frustrating task, in absence of a widely agreed upon standard formatting :slightly_smiling_face:

Comment (risicle) [reply pkgs/applications/networking/cluster/nomad/genericModule.nix]: Agreed 1000x over

**Comment by zowoq** [comment]
Can we move both to `buildGoModule` so we don't need to keep the two `generic.nix` implementations in sync?

```diff
diff --git a/pkgs/applications/networking/cluster/nomad/1.0.nix b/pkgs/applications/networking/cluster/nomad/1.0.nix
index 4966c42b9a4..63dcb455753 100644
--- a/pkgs/applications/networking/cluster/nomad/1.0.nix
+++ b/pkgs/applications/networking/cluster/nomad/1.0.nix
@@ -1,11 +1,12 @@
 { callPackage
-, buildGoPackage
+, buildGoModule
 , nvidia_x11
 , nvidiaGpuSupport
 }:
 
 callPackage ./generic.nix {
-  inherit buildGoPackage nvidia_x11 nvidiaGpuSupport;
+  inherit buildGoModule nvidia_x11 nvidiaGpuSupport;
   version = "1.0.9";
   sha256 = "0ml6l5xq1310ib5zqfdwlxmsmhpc5ybd05z7pc6zgxbma1brxdv4";
+  vendorSha256 = null;
 }
diff --git a/pkgs/applications/networking/cluster/nomad/1.1.nix b/pkgs/applications/networking/cluster/nomad/1.1.nix
index e9c4ff8c967..fa154411a00 100644
--- a/pkgs/applications/networking/cluster/nomad/1.1.nix
+++ b/pkgs/applications/networking/cluster/nomad/1.1.nix
@@ -4,7 +4,7 @@
 , nvidiaGpuSupport
 }:
 
-callPackage ./genericModule.nix {
+callPackage ./generic.nix {
   inherit buildGoModule nvidia_x11 nvidiaGpuSupport;
   version = "1.1.3";
   sha256 = "0jpc8ff56k9q2kv9l86y3p8h3gqbvx6amvs0cw8sp4i7dqd2ihz2";
diff --git a/pkgs/applications/networking/cluster/nomad/generic.nix b/pkgs/applications/networking/cluster/nomad/generic.nix
index 99a2585272c..224b6487357 100644
--- a/pkgs/applications/networking/cluster/nomad/generic.nix
+++ b/pkgs/applications/networking/cluster/nomad/generic.nix
@@ -1,26 +1,26 @@
 { lib
-, buildGoPackage
+, buildGoModule
 , fetchFromGitHub
 , version
 , sha256
+, vendorSha256
 , nvidiaGpuSupport
 , patchelf
 , nvidia_x11
 , nixosTests
 }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "nomad";
-  inherit version;
-  rev = "v${version}";
+  inherit version vendorSha256;
 
-  goPackagePath = "github.com/hashicorp/nomad";
   subPackages = [ "." ];
 
   src = fetchFromGitHub {
     owner = "hashicorp";
     repo = pname;
-    inherit rev sha256;
+    rev = "v${version}";
+    inherit sha256;
   };
 
   nativeBuildInputs = lib.optionals nvidiaGpuSupport [
@@ -45,7 +45,6 @@ buildGoPackage rec {
   meta = with lib; {
     homepage = "https://www.nomadproject.io/";
     description = "A Distributed, Highly Available, Datacenter-Aware Scheduler";
-    platforms = platforms.unix;
     license = licenses.mpl20;
     maintainers = with maintainers; [ rushmorem pradeepchhetri endocrimes maxeaubrey ];
   };
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 1ae36ce6407..60b02c98aaa 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7435,7 +7435,7 @@ in
   # Upstream partially documents used Go versions here
   # https://github.com/hashicorp/nomad/blob/master/contributing/golang.md
   nomad_1_0 = callPackage ../applications/networking/cluster/nomad/1.0.nix {
-    buildGoPackage = buildGo115Package;
+    buildGoModule = buildGo115Module;
     inherit (linuxPackages) nvidia_x11;
     nvidiaGpuSupport = config.cudaSupport or false;
   };
```

**Comment by maxeaubrey** [comment]
I don't have a particular desire to change the build process for `1.0`, as I'd like to maintain the ability to easily backport those version bumps for the remainder of this cycle. I assume/hope that by 21.11 we'll be dropping 1.0 anyway, but if not, I'd rather switch all nomad versions to use `buildGoModule` only at that point.

**Comment by zowoq** [comment]
If we aren't going to move both to `buildGoModule` please fix `passthru.tests` and `meta.platforms` which have diverged between `1.0` and `1.1`.

**Review by endocrimes — APPROVED** [review]

**Pull request merged**